apps/studio/src/pages/sites/[siteId]/pages/[pageId]/index.tsx

```diff
@@ -117,148 +122,169 @@ const PageEditingView = ({ page, permalink, siteId }: PageEditingViewProps) => {
   )
 }
 
-const pageSettingsSchema = z.object({
-  title: pageTitleSchema,
-  meta: z.string().optional(),
-  permalink: permalinkSchema,
-  noIndex: z.boolean().optional(),
-})
+const THREE_SECONDS = 3000
+const SUCCESS_TOAST_ID = "save-page-settings-success"
 
 interface PageSettingsProps {
   permalink: RouterOutput["page"]["readPageAndBlob"]["permalink"]
-  page: RouterOutput["page"]["readPageAndBlob"]["content"]
+  title: RouterOutput["page"]["readPageAndBlob"]["title"]
+  type: RouterOutput["page"]["readPageAndBlob"]["type"]
 }
 const PageSettings = ({
   permalink: originalPermalink,
-  page,
+  type,
+  title: originalTitle,
 }: PageSettingsProps) => {
-  const { register, setValue, getFieldState, watch, control } = useZodForm({
-    schema: pageSettingsSchema,
+  const { pageId, siteId } = useQueryParse(editPageSchema)
+  const { register, watch, control, handleSubmit } = useZodForm({
+    schema: pageSettingsSchema.omit({ pageId: true, siteId: true }),
     defaultValues: {
-      title: page.page.title || "",
+      title: originalTitle || "",
       permalink: originalPermalink || "",
     },
   })
 
   const [title, permalink] = watch(["title", "permalink"])
 
-  useEffect(() => {
-    const permalinkFieldState = getFieldState("permalink")
-    // This allows the syncing to happen only when the page title is not dirty
-    // Dirty means user has changed the value AND the value is not the same as the default value of "".
-    // Once the value has been cleared, dirty state will reset.
-    if (!permalinkFieldState.isDirty) {
-      setValue("permalink", generateResourceUrl(title), {
-        shouldValidate: !!title,
+  const toast = useToast({ duration: THREE_SECONDS, isClosable: true })
+  const utils = trpc.useUtils()
+
+  const updatePageSettingsMutation = trpc.page.updatePageSettings.useMutation({
+    onSuccess: async () => {
+      // TODO: we should use a specialised query for this rather than the general one that retrives the page and the blob
+      await utils.page.readPageAndBlob.invalidate()
+      await utils.page.readPage.invalidate()
+      if (!toast.isActive(SUCCESS_TOAST_ID)) {
+        toast({
+          id: "save-page-settings-success",
+          title: "Saved page settings",
+          description: "Publish this page for your changes to go live.",
+          status: "success",
+        })
+      }
+    },
+    onError: (error) => {
+      toast({
+        title: "Failed to save page settings",
+        description: error.message,
+        status: "error",
       })
-    }
-  }, [getFieldState, setValue, title])
+    },
+  })
+
+  const onSubmit = handleSubmit(async (values) => {
+    await updatePageSettingsMutation.mutateAsync({ pageId, siteId, ...values })
```

Comment:
[Prefer mutate over mutateAsync](https://www.notion.so/opengov/Prefer-mutate-over-mutateAsync-ac450f152af84ec880ab520ca1520ebd)

Reply:
might also want to update the latest rhf state so this does not run if there are no changes. You may want to call `reset` with the values you get here. Then check `isDirty` form state before triggering the mutation, perhaps?

Reply:
i did the `isDirty` check, you're right! not too sure on the `reset` part - the reason why is when we submit, we invalidate the cache and cause a refetch which trickles down to this component as `originalPermalink, originalTitle`, which is used as `defaultValues` 

is the `reset` still required here? i'm assuming rhf will use the new values of `originalPermalink` and `originalTitle`

(also made the change to use `mutate`, i don't recall why i used `mutateAsync` honestly)

Reply:
If the form correctly resets with the updated values then sure can skip reset haha

Reply:
ah dang, it doesn't quite get the values correctly. oddly enough, it's still using the old values for the `isDirty` comparison (not very sure why, do you know?) and as a result, i reset the form values now with the `mutate`'s `onSuccess` hook

Reply:
Because the default value of the form hasn't changed when you submit. See [docs](https://react-hook-form.com/docs/useformstate#:~:text=Description-,isDirty,-boolean). rhf's [`reset`](https://react-hook-form.com/docs/useform/reset) will update the default values to the new input (which is the updated values in this case)

Reply:
i'm abit confused here - after submit, wouldn't the (old) value be invalidated and thus trigger a refetch from db? then, this would cause new data to be sent down to this component and then, get set as the new `defaultValue`?

Reply:
No, cause the value is cached in rhf. You need to remount the component to use the new prop in that case. You can imagine if the defaultValues are set in a useState in the start. New props won't update the default values since they've already been set in a useState.


```ts
// First render, propDefaultValues = 1
const [defaultValues, _setDefaultValues] = useState(propDefaultValues)
// defaultValues === 1)

// Second render, propDefaultValues = 2
// defaultValues already has a value, nothing will happen. useState will not rerun since the component has not remounted (lifecycle)
// defaultValues still === 1
```